test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/WpfTestCase.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Threading;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace NuGet.PackageManagement.UI.Test
+{
+    [DebuggerDisplay(@"\{ class = {TestMethod.TestClass.Class.Name}, method = {TestMethod.Method.Name}, display = {DisplayName}, skip = {SkipReason} \}")]
+    public sealed class WpfTestCase : LongLivedMarshalByRefObject, IXunitTestCase
+    {
+        private IXunitTestCase _testCase;
+
+        public string DisplayName => _testCase.DisplayName;
+        public IMethodInfo Method => _testCase.Method;
+        public string SkipReason => _testCase.SkipReason;
+        public ITestMethod TestMethod => _testCase.TestMethod;
+        public object[] TestMethodArguments => _testCase.TestMethodArguments;
+        public Dictionary<string, List<string>> Traits => _testCase.Traits;
+        public string UniqueID => _testCase.UniqueID;
+
+        public ISourceInformation SourceInformation
+        {
+            get { return _testCase.SourceInformation; }
+            set { _testCase.SourceInformation = value; }
+        }
+
+        public WpfTestCase(IXunitTestCase testCase)
+        {
+            if (testCase == null)
+            {
+                throw new ArgumentNullException(nameof(testCase));
+            }
+
+            _testCase = testCase;
+        }
+
+        [Obsolete("Called by the deserializer", error: true)]
+        public WpfTestCase() { }
+
+        public Task<RunSummary> RunAsync(IMessageSink diagnosticMessageSink,
+            IMessageBus messageBus,
+            object[] constructorArguments,
+            ExceptionAggregator aggregator,
+            CancellationTokenSource cancellationTokenSource)
+        {
+            var taskCompletionSource = new TaskCompletionSource<RunSummary>();
+            var thread = new Thread(() =>
```

Comment:
Yes, it creates a new thread per test.  Execution for all 13 tests takes ~4.5 seconds on my dev VM.